pkg/aggregator/internal/limiter/config.go

```diff
@@ -0,0 +1,47 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package limiter
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// FromConfig builds new Limiter from the configuration.
+func FromConfig(pipelineCount int) *Limiter {
+	return fromConfig(pipelineCount, getCgroupMemoryLimit)
+}
+
+func fromConfig(pipelineCount int, cgroupLimitGetter func() (uint64, error)) *Limiter {
+	limit := 0
+	memoryRatio := config.Datadog.GetFloat64("dogstatsd_context_limiter.cgroup_memory_ratio")
+	bytesPerContext := config.Datadog.GetInt("dogstatsd_context_limiter.bytes_per_context")
+	if memoryRatio > 0 && bytesPerContext > 0 {
+		cgroupLimit, err := cgroupLimitGetter()
+		if err != nil {
+			log.Errorf("dogstatsd context limiter: memory based limit configured, but: %v", err)
+		} else {
+			limit = int(memoryRatio*float64(cgroupLimit)) / bytesPerContext
+			log.Debugf("dogstatsd context limiter: memory limit=%d, ratio=%f, contexts limit=%d", cgroupLimit, memoryRatio, limit)
+		}
+	}
+
+	if limit == 0 {
+		limit = config.Datadog.GetInt("dogstatsd_context_limiter.limit")
```

Comment:
Essentially this is declaring a precedence of the memory ratio limiting _over_ the fixed limit.
So if all of the following are true:
1. `dogstatsd_context_limiter.cgroup_memory_ratio` is set to a valid value
2. `dogstatsd_context_limiter.bytes_per_context` is set to a valid value
3. no errors occur while fetching cgroup limit

Then the mem ratio based limiting will apply.
Else the static limit defined by `dogstatsd_context_limiter.limit` will be used.

Is this correct? If so, can we document this behavior somewhere?

Reply:
Added a comment to the function, does it address you concern?